### PR TITLE
feat(welcome/native): settings updates

### PR DESCRIPTION
**Pull request by Calinteodor**
1. Removed Drawer from Welcome page and it's dependencies.
2. Removed re-animated dependency.
3. Added settings tab on Welcome page.
4. Created SettingsButton inside OverflowMenu.
5. Removed HelpButton from the OverflowMenu.

### CODE REVIEW

File: react/features/mobile/navigation/components/RootNavigationContainer.js [hunk 1]
```diff
@@ -63,15 +68,36 @@ const RootNavigationContainer = ({ dispatch, isWelcomePageAvailable }: Props) =>
...
+                                options = { welcomeScreenOptions } />
                             <RootStack.Screen
                                 component = { DialInSummary }
                                 name = { screen.dialInSummary }
                                 options = { dialInSummaryScreenOptions } />
                         </>
                 }
+                <RootStack.Screen
+                    component = { HelpView }
+                    name = { screen.welcome.tabs.settings.help }
```
Comment (saghul) [inline react/features/mobile/navigation/components/RootNavigationContainer.js:82]: Maybe we should remove "welcome" from the name here, since it no longer applies just to the welcome page.

File: react/features/mobile/navigation/components/welcome/functions.js [hunk 2]
```diff
@@ -18,7 +17,6 @@ import { navigationStyles } from '../styles';
 export function renderArrowBackButton(onPress: Function) {
     return (
         <HeaderNavigationButton
-            buttonStyle = { navigationStyles.arrowBackStyle }
```
Comment (saghul) [inline react/features/mobile/navigation/components/welcome/functions.js:21]: Why is this no longer necessary?

File: react/features/mobile/navigation/routes.js [hunk 3]
```diff
@@ -1,15 +1,18 @@
...
         tabs: {
             recent: 'Recent',
-            calendar: 'Calendar'
-        },
-        settings: 'Settings',
-        terms: 'Terms',
-        privacy: 'Privacy',
-        help: 'Help'
+            calendar: 'Calendar',
+            settings: {
```
Comment (saghul) [inline react/features/mobile/navigation/routes.js:9]: Let's move this to the top level, since this is a top level screen now.

File: react/features/welcome/components/native/settings/components/SettingsView.js [hunk 4]
```diff
@@ -200,6 +199,20 @@ class SettingsView extends AbstractSettingsView<Props, State> {
             }
         };
 
+        let helpScreenRoute;
+        let termsScreenRoute;
+        let privacyScreenRoute;
+
+        if (this.props.isTabNavigatorScreen) {
```
Comment (saghul) [inline react/features/welcome/components/native/settings/components/SettingsView.js:206]: Why is this necessary? Should we make the screen a container itself and navigate to those routes within?

File: react/features/welcome/components/native/settings/components/SettingsView.js [hunk 5]
```diff
@@ -595,4 +630,21 @@ function _mapStateToProps(state) {
...
+
+    useEffect(() => {
+        if (isTabNavigatorScreen) {
+            if (isFocused) {
+                onSettingsScreenFocused(true);
+            } else {
+                onSettingsScreenFocused(false);
+            }
+        }
+    });
```
Comment (saghul) [inline react/features/welcome/components/native/settings/components/SettingsView.js:645]: Shouldn't we filter by isTabNavigatorScreen and isFocused ?

File: react/features/welcome/constants.js [hunk 6]
```diff
@@ -37,3 +37,12 @@ export const calendarListTabBarOptions = {
             src = { IconEventNote } />
     )
 };
+
+export const settingsTabBarOptions = {
+    // $FlowExpectedError
```
Comment (saghul) [inline react/features/welcome/constants.js:42]: Drop the flow comment.

**Review by saghul — COMMENTED** [review]
Did a round of code review, I'll take it for a spin now.

**Comment by saghul** [comment]
Some UI / UX comments after testing:

- The room input change you made seems to work great!
- Don't make the links section foldable, let's have it always expanded
- Each link "row" should be as tall as the other rows
- It's weird that the screen behaves differently while in a conference. I think Settings should become its own navigator with the main and links screens. This also paves the way for the future, when we add more settings sections. Then none of the screens inside settings would be presentation modals, so the back arrow makes sense.

File: react/features/welcome/components/native/settings/components/SettingsView.js (hunk 5 above)
Comment (Calinteodor) [reply react/features/welcome/components/native/settings/components/SettingsView.js:645]: We can do that yes.

File: react/features/welcome/components/native/settings/components/SettingsView.js (hunk 4 above)
Comment (Calinteodor) [reply react/features/welcome/components/native/settings/components/SettingsView.js:206]: SettingsNavigationContainer can be used as a container screen only inside ConferenceNavigationContainer. 
If we use it inside the WelcomePageTabs it kind of complicates thing with navigation and screen options.

File: react/features/mobile/navigation/components/RootNavigationContainer.js (hunk 1 above)
Comment (Calinteodor) [reply react/features/mobile/navigation/components/RootNavigationContainer.js:82]: I suggest we leave it as it is because this is part of a different navigator.

File: react/features/mobile/navigation/components/welcome/functions.js (hunk 2 above)
Comment (Calinteodor) [reply react/features/mobile/navigation/components/welcome/functions.js:21]: Based on visual testing, this is no longer necessary.

File: react/features/welcome/components/native/settings/components/SettingsView.js (hunk 5 above)
Comment (Calinteodor) [reply react/features/welcome/components/native/settings/components/SettingsView.js:645]: I think it better if we keep this way because the most we have to check first if it's a tab screen or a navigator screen. 
Afterwards we apply the focus logic.

**Comment by Calinteodor** [comment]
> Some UI / UX comments after testing:
> 
> * The room input change you made seems to work great!
> * Don't make the links section foldable, let's have it always expanded
> * Each link "row" should be as tall as the other rows
> * It's weird that the screen behaves differently while in a conference. I think Settings should become its own navigator with the main and links screens. This also paves the way for the future, when we add more settings sections. Then none of the screens inside settings would be presentation modals, so the back arrow makes sense.

I agree and I created a SettingsNavigationContainer but that can onl

File: react/features/mobile/navigation/routes.js (hunk 3 above)
Comment (Calinteodor) [reply react/features/mobile/navigation/routes.js:9]: I created a diferent route obj for SettingsNavigationContainer.

**Comment by saghul** [comment]
Why are we not rendering the Settings container in the welcome page?

**Comment by saghul** [comment]
Not sure what this means, can you clarify?

**Comment by Calinteodor** [comment]
> Not sure what this means, can you clarify?

No need. You can the PR for another spin.

File: react/features/mobile/navigation/components/settings/components/SettingsNavigationContainer.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+type Props = {
+
+    /**
+     * Callback to be invoked when settings screen is focused.
+     */
+    onSettingsScreenFocused: Function
+};
+
+
+const SettingsNavigationContainer = ({ onSettingsScreenFocused }: Props) => {
```
Comment (saghul) [inline react/features/mobile/navigation/components/settings/components/SettingsNavigationContainer.js:37]: Please rename `onSettingsScreenFocused` to `isInWelcomePage` (and make it a boolean) and we're good!

Comment (Calinteodor) [reply react/features/mobile/navigation/components/settings/components/SettingsNavigationContainer.js]: Done.

File: react/features/mobile/navigation/components/welcome/components/WelcomePageTabs.js [hunk 8]
```diff
@@ -28,42 +30,80 @@ type Props = {
...
+        (
+            <CalendarList
+                disabled = { disabled } />
+        )
+    );
+
+    const SettingsScreen = useCallback(() =>
+        (
+            <SettingsNavigationContainer
+                isInWelcomePage = { onSettingsScreenFocused } />
```
Comment (saghul) [inline react/features/mobile/navigation/components/welcome/components/WelcomePageTabs.js:63]:
```suggestion
                isInWelcomePage = { true } />
```

**Review by saghul — APPROVED** [review]
